pkg/cloud/libvirt/libvirt.go

```diff
@@ -0,0 +1,118 @@
+package libvirt
+
+import (
+	"fmt"
+	"github.com/osbuild/images/pkg/cloud"
+	"io"
+	"os"
+
+	lv "libvirt.org/go/libvirt"
+)
+
+type libvirtUploader struct {
+	connection string
+	pool       string
+	volume     string
+}
+
+func NewUploader(connection string, pool string, volume string) (cloud.Uploader, error) {
+	return &libvirtUploader{
+		connection: connection,
+		pool:       pool,
+		volume:     volume,
+	}, nil
+}
+
+func (lu *libvirtUploader) Check(status io.Writer) error {
+	return nil
+}
+
+func (lu *libvirtUploader) UploadAndRegister(r io.Reader, f *os.File, status io.Writer) (err error) {
+	fmt.Fprintf(status, "Uploading to libvirt...\n")
+
+	// Connect to libvirt
+	conn, err := lv.NewConnect(lu.connection)
+	if err != nil {
+		return fmt.Errorf("Failed to connect to libvirt: %v", err)
```

Comment:
Please use `%w` when wrapping errors, that ensures that the upper layers of go can use things like `errors.Is` to unwrap the underlying error type (here and below).